src/lib/macros.h

```diff
@@ -46,4 +48,19 @@
 		_ptr = NULL; \
 	} while (0)
 
+/**
+ * @brief allocate memory of indicated size and zero it.
+ *
+ * Abort on out of memory errors.
+ */
+static inline void *malloc_or_die(size_t size)
```

Comment:
I find this name a little confusing because I would have assumed you were using malloc within this function. Do you think is worth using a different word to avoid this confusion? why not naming it calloc _or_die if the function is really using calloc?